atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+                else:
+                    param_dict[key] = self._data_parser.get(path_key)
+                    if isinstance(param_dict[key], str):
+                        if key == 'thermodynamic_ensemble':
+                            param_dict[key] = param_dict[key].upper()  # TODO change enums to lower case and adjust Gromacs and Lammps code accordingly
+                        else:
+                            param_dict[key] = param_dict[key].lower()
+                    elif isinstance(param_dict[key], (int, np.int32, np.int64)):
+                        param_dict[key] = param_dict[key].item()
+
+            return param_dict
+
+        path_force_calculations = 'parameters.force_calculations'
+        force_calculations_group = self._data_parser.get(path_force_calculations)
+        if force_calculations_group is not None:
+            self._parameter_info['force_calculations'] = get_parameters(force_calculations_group, path_force_calculations)
+        path_workflow = 'parameters.workflow'
+        workflow_group = self._data_parser.get(path_workflow)
+        if workflow_group is not None:
+            self._parameter_info['workflow'] = get_parameters(workflow_group, path_workflow)
+
+    def parse_calculation(self):
+        sec_run = self.archive.run[-1]
+        calculation_info = self._observable_info.get('configurational')
+        system_info = self._system_info.get('calculation')  # note: it is currently ensured in parse_system() that these have the same length as the system_map
+        if not calculation_info:  # TODO should still create entries for system time link in this case
+            return
+
+        for step in self.steps:
+            data = {
+                'method_ref': sec_run.method[-1] if sec_run.method else None,
+                'step': step,
+                'energy': {},
+            }
+            data_h5md = {
+                'x_h5md_custom_calculations': [],
+                'x_h5md_energy_contributions': []
+            }
+            data['time'] = calculation_info.get('step', {}).get('time') # * self._time_unit
+            if not data['time']:
+                data['time'] = system_info.get('step', {}).get('time')
+
+            if system_info.get(step):
+                for key, val in system_info[step].items():
+                    if key == 'forces':
+                        data[key] = dict(total=dict(value=val))
+                    else:
+                        if hasattr(BaseCalculation, key):
+                            data[key] = val
+                        else:
+                            unit = None
+                            if hasattr(val, 'units'):
+                                unit = val.units
+                                val = val.magnitude
+                            data_h5md['x_h5md_custom_calculations'].append(CalcEntry(kind=key, value=val, unit=unit))
+
+            if calculation_info.get(step):
```

Comment:
for key, val in calculation_info.get(step, {})

Reply:
done.